components/utils/DataProvider.js

```diff
@@ -13,34 +15,87 @@ class DataProvider extends Component {
       networks: ["Testnet One", "Testnet Two"],
       currentNetwork: null,
       updateNetwork: this.updateNetwork,
-      details: dummyData.details,
-      blocks: dummyData.blocks,
+      details: this.props.details,
+      blocks: this.props.blocks,
       transactions: dummyData.transactions
     };
   }
 
-  componentWillMount() {
+  static async getInitialProps(ctx) {
+    const [details, blocks] = await Promise.all([
+      DataProvider.getDetails(),
+      DataProvider.getBlocksInfo()
+    ]);
+    return {
+      details,
+      blocks
+    };
+  }
+
+  componentDidMount() {
     this.updateNetwork(0);
   }
 
+  static async getBlocksInfo() {
+    let blocks;
+    try {
+      blocks = await call(".select", [
```

Comment:
Reaching the backend from the frontend is as simple as `call('remote-function-name', ['params'])`.